### PR TITLE
TELCODOCS-1050: Configuring the hub cluster to use unauthenticated registries

### PR DESCRIPTION
**Pull request by kdrake20**
Fixes: [TELCODOCS-1050](https://issues.redhat.com/browse/TELCODOCS-1050)

For: Version 4.12+

Doc Preview [here](https://55352--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.html#ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries_ztp-preparing-the-hub-cluster)

Signed-off-by: Katie Tothill ktothill@redhat.com

### CODE REVIEW

File: modules/proc_ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (aireilly) [inline modules/proc_ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:2]: Suggest renaming the file to match the id, so `modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc`.

**Comment by ktothill** [comment]
> Suggest renaming the file to match the id, so `modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc`. 

Updated as suggested.

File: modules/proc_ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (ktothill) [reply modules/proc_ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc]: Updated as suggested.

File: scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc
Comment (aireilly) [inline scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc:37]:
```suggestion
* xref:../../scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc#ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries_ztp-preparing-the-hub-cluster[Configuring the hub cluster to use unauthenticated registries]
```

Comment (ktothill) [reply scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc]: Updated as suggested.

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (aireilly) [inline modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:4]: Lose this! Only set `:context:` in assemblies :)

Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc]: Updated as suggested.

File: scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc
Comment (aireilly) [inline scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc:37]:
```suggestion
* xref:../scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc#ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries_ztp-preparing-the-hub-cluster[Configuring the hub cluster to use unauthenticated registries]
```

**Review by aireilly — COMMENTED** [review]
xref is broken. See fix below.

File: scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc
Comment (ktothill) [reply scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc]: Updated as suggested.

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (amolnar-rh) [inline modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:369]: To generate a link here, you have to use the following syntax:
`link:<https://example.com>[Example Title]`

(Btw, I think I see the same problem on [line 179](https://github.com/openshift/openshift-docs/pull/55352/files#diff-409f6af30b90a6d04d6ba57f8de6ffe514553beb7f7c14717e92bd09dcc614e5R179))

Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc]: Updated as suggested.

**Comment by ktothill** [comment]
*RN DRAFT*

[id='id_{context}']
= Configuring the hub cluster to use unauthenticated registries
This release supports the use of unauthenticated registries when configuring the hub cluster. Registries that don't require authentication can be listed under `spec.unauthenticatedRegistries` in the `AgentServiceConfig` resource. Any registry on this list is not required to have an entry in the pull secret validated by the `assisted-service`.

For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc#ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries_ztp-preparing-the-hub-cluster

**Comment by ktothill** [comment]
/peer-review-needed

**NOTE:** The requested review is only for the RN. The content is still WIP. Thank you!

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (CrystalChun) [inline modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:17]: Sorry, where in the official docs will this be located? The reason I ask is because the instructions for this procedure below directly refers to the assisted-service github repo

Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:17]: Hi, @CrystalChun. The content is going in Scalability and Performance after [Configuring the hub cluster to use a disconnected mirror registry](https://docs.openshift.com/container-platform/4.12/scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.html#ztp-configuring-the-cluster-for-a-disconnected-environment_ztp-preparing-the-hub-cluster) as part of 4.12.z. Thanks!

**Comment by trewest** [comment]
Hi @ktothill I'm a bit confused about everything in the ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries_ztp-preparing-the-hub-cluster section besides the unauthenticatedRegistries part. unauthenticatedRegistries is just an optional field in the AgentServiceConfig and it seems like there is already information in the doc regarding the agentserviceconfig

**Comment by ktothill** [comment]
@trewest thanks so much for the question! This is just an update to docset to be placed after [Configuring the hub cluster to use a disconnected mirror registry](https://docs.openshift.com/container-platform/4.12/scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.html#ztp-configuring-the-cluster-for-a-disconnected-environment_ztp-preparing-the-hub-cluster).

Here is the Jira story, which may help clarify:
https://issues.redhat.com/browse/TELCODOCS-1050

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (trewest) [inline modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:25]: I would clarify that it is the pull secret created for the spoke cluster

**Comment by trewest** [comment]
@ktothill I think what you have included in this PR makes sense for the most part regarding the unauthenticatedRegistries feature. My concern is with the other content in the doc preview in the beginning of the `Cofiguring the hub cluster to use unauthenticated registries` section, before the pieces you have added. These are steps for deploying the community version of the assisted-service operator and include material from the github project. It doesn't represent a supported method for deploying the operator.

**Comment by ktothill** [comment]
@trewest thanks for picking up on that! I've updated the assembly to reflect the correct placement.

**Comment by ktothill** [comment]
UPDATED RN DRAFT

[id="ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries"]
= Configuring the hub cluster to use unauthenticated registries
This release supports the use of unauthenticated registries when configuring the hub cluster. Registries that don't require authentication can be listed under spec.unauthenticatedRegistries in the AgentServiceConfig resource. Any registry on this list is not required to have an entry in the pull secret validated by the assisted-service.

For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc#ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries_ztp-preparing-the-hub-cluster

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (CrystalChun) [inline modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:22]: It might be a good idea to match the format of this procedure https://55352--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.html#enabling-assisted-installer-service-on-bare-metal_ztp-preparing-the-hub-cluster

```suggestion
. Update the AgentServiceConfig CR by running the following command:

[source,terminal]
----
$ oc edit AgentServiceConfig
---

. Add the `unauthenticatedRegistries` field in the CR:

[source,terminal]
----
apiVersion: agent-install.openshift.io/v1beta1
kind: AgentServiceConfig
metadata:
  name: agent
spec:
  unauthenticatedRegistries:
  - example.registry.com
  - example.registry2.com
  ...
----
```

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (CrystalChun) [inline modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:32]: Likewise for these lines, the `[Mirror registries](#mirror-registry-configuration)` and `[ConfigMap override](#specifying-environmental-variables-via-configmap)` are leftover from a different document. Perhaps edit the linkage or remove the link altogether.

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (aireilly) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:22]: +1 please make this change

**Review by aireilly — COMMENTED** [review]
Added some comments

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:22]: Updated as suggested.

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:26]: Updated as suggested.

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:32]: Updated as suggested.

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:6]: Not that I'm aware of. @trewest, perhaps you could confirm if there are any other prereqs (network)?

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:25]: Updated as suggested.

**Comment by ktothill** [comment]
@trewest @CrystalChun I've made the requested updates. Please review and let me know if any changes are needed. Thank you!!

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (trewest) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:6]: No I don't think any other prerequisites should be included. The cluster must be able to reach the registry over the network but I think that is common knowledge when working with registries.

**Review by CrystalChun — APPROVED** [review]
/lgtm

File: scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc
Comment (aireilly) [inline scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc:43]: Why have you removed this xref?

Comment (aireilly) [reply scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc:43]: @ktothill I've made this change and merged a PR which fixes the build error. Please update your local copy :)

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (aireilly) [inline modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:33]:
```suggestion
[source,yaml]
```

**Comment by trewest** [comment]
@CrystalChun should we include a comment with `PUBLIC_CONTAINER_REGISTRIES` defaults (i.e. quay.io, registry.svc.ci.openshift.org)?

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:33]: Updated as suggested.

File: scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc
Comment (ktothill) [reply scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc:43]: Updated as suggested.

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (trewest) [inline modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:28]: This should technically be 
```
oc edit AgentServiceConfig agent
```

File: scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc
Comment (aireilly) [reply scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc:43]: Did you rebase your local copy like I suggested? you've reintroduced the error I fixed.

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (aireilly) [inline modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:29]:
```suggestion
----
```
Top and bottom codeblock delimiters need to match otherwise the build get's messed up.

File: scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc
Comment (aireilly) [inline scalability_and_performance/ztp_far_edge/ztp-preparing-the-hub-cluster.adoc:42]: Remove these lines, you've deleted the xref so they serve no purpose.

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:28]: Updated as suggested.

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:29]: Updated as suggested.

**Comment by ktothill** [comment]
@trewest @CrystalChun I've made the additional requested updates. 

Please review and provide an `LGTM` let me know if any changes are needed. Thank you!!

**Comment by ktothill** [comment]
/label peer-review-needed

Requesting an updated peer review to confirm that the content and RN are ready for release.

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (sheriff-rh) [inline modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:18]: `cluster-admin` in backticks

**Review by sheriff-rh — COMMENTED** [review]
Looks good, Katie! One small nit :)

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:18]: Updated as suggested.

**Comment by aireilly** [comment]
@trewest I think it would be good to add a verification procedure that the user can do to verify that the unauthenticated registry is accesible, for example, what about: 

```asciidoc
.Verification

Verify that you can access the newly added registry from the hub cluster by running the following commands:

. Open a debug shell to the hub cluster:
+
[source,terminal]
----
$ oc debug node/<node_name>
----

. Test access to the unauthenticated registry by running the following command:
+
[source,terminal]
----
sh-4.4# podman login -u kubeadmin -p $(oc whoami -t) unauthenticated-image-registry.openshift-image-registry.svc:5000
----
+
.Example output
[source,terminal]
----
Login Succeeded!
----
```

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (CrystalChun) [inline modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc:54]: > @CrystalChun should we include a comment with `PUBLIC_CONTAINER_REGISTRIES` defaults (i.e. quay.io, registry.svc.ci.openshift.org)?

Agreed, @ktothill please include these defaults. I think here would be best, wdyt @ktothill @trewest?

Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc]: I can add those defaults. Are there any others outside of [quay.io](http://quay.io/), [registry.svc.ci.openshift.org](http://registry.svc.ci.openshift.org/)?

**Comment by skrthomas** [comment]
@ktothill Just want to confirm, this does not need peer review, correct? It has both labels, so I'm unsure. Same goes for the RN for this.

**Review by snarayan-redhat — COMMENTED** [review]
Assuming that you needed another round of peer review after you pushed the changes.

LGTM!

File: modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc
Comment (ktothill) [reply modules/ztp-configuring-the-hub-cluster-to-use-unauthenticated-registries.adoc]: Added default values as suggested.

**Comment by ktothill** [comment]
@trewest, could I please get an updated `LGTM` from you and @danielerez (I need SME and QE `LGTM` to merge)? Thank you!!

**Comment by ktothill** [comment]
@trewest and @danielerez  could I please get an updated `LGTM` (I need SME and QE LGTM to merge)? Thank you!!

**Pull request merged**
